explainaboard/processors/conditional_generation.py

```diff
@@ -427,6 +434,10 @@ def get_bucket_performance_tok(
                 toks_pred = samples_over_bucket_pred[bucket_interval]
 
             p_denom, r_denom = len(toks_pred), len(toks_true)
+
+            # TODO(odashi): I didn't understand what these lines are doing.
+            # These lines clearly violate type hints as far as I believed those in
+            # the argument list, but I couldn't figure out the correct typing.
```

Comment:
@neubig please give me details about this.

Reply:
The four dimensional data structure is saying:
* What sentence is it? `x[0]`
* What variety of information about the sentence? `hyp_tok_info`: information about the hypothesis tokens.
* What token is it? `x[1]`
* What variety of information about they hypothesis token? `tok_matched`: whether or not the token matches a token in the reference.

And analogously for the reference tokens, whether they match the hypothesis.

Reply:
so does the code here intend `dict[(int, str, int, str) -> something]`? as `x` has the type of `str`, the first and third subscription is warned as type mismatch. The type information here is inherited from the function signature and I couldn't figure out any quick fix on this function.

Reply:
@neubig Could you separate the appropriate part into another pr? I can try to merge it anyway but it requires certain cherry-picking maybe

Reply:
let's merge this mypy PR and I'll fix the merge errors later